pkg/backup/backup.go

```diff
@@ -170,8 +169,22 @@ func NewKubernetesBackupper(
 
 // getNamespaceIncludesExcludes returns an IncludesExcludes list containing which namespaces to
 // include and exclude from the backup.
-func getNamespaceIncludesExcludes(backup *velerov1api.Backup) *collections.IncludesExcludes {
-	return collections.NewIncludesExcludes().Includes(backup.Spec.IncludedNamespaces...).Excludes(backup.Spec.ExcludedNamespaces...)
+func getNamespaceIncludesExcludes(backup *velerov1api.Backup, kbClient kbclient.Client) (*collections.NamespaceIncludesExcludes, []string, error) {
+	includesExcludes := collections.NewNamespaceIncludesExcludes().Includes(backup.Spec.IncludedNamespaces...).Excludes(backup.Spec.ExcludedNamespaces...)
+	nsList := corev1api.NamespaceList{}
+	activeNamespaces := []string{}
+	nsManagedByArgoCD := []string{}
+	if err := kbClient.List(context.Background(), &nsList); err != nil {
+		return nil, nsManagedByArgoCD, err
+	}
+	for _, ns := range nsList.Items {
+		activeNamespaces = append(activeNamespaces, ns.Name)
+		nsLabels := ns.GetLabels()
+		if len(nsLabels[ArgoCDManagedByNamespaceLabel]) > 0 && includesExcludes.ShouldInclude(ns.Name) {
+			nsManagedByArgoCD = append(nsManagedByArgoCD, ns.Name)
+		}
+	}
+	return includesExcludes.ActiveNamespaces(activeNamespaces), nsManagedByArgoCD, nil
```

Comment:
Why do we need to change this behavior? --- return the full ns list
I don't see anywhere is using the full ns list necessarily.

Reply:
We need the full namespace list to compare overlap. The refactoring here for NamespaceIncludesExcludes does 2 things:
1) Provides a convenient way of getting a full list of included namespaces in the backup (needed for overlap detection)
2) Provides a common location for implementing the wildcard support that @Joeavaikath is working on.